frontend/src/components/app/Modal.js

```diff
@@ -80,10 +89,58 @@ class Modal extends Component {
     this.mounted = false;
 
     this.removeEventListeners();
+
+    // disconnect when the component is unmounted
+    this.modalWsClient &&
+      this.modalWsClient.connected &&
+      this.modalWsClient.disconnect();
   }
 
+  /**
+   * @method initModalWsConnection
+   * @summary - connect and subscribes to the subscription `websocket` topic (websocket endpoint)
+   * @param {string} websocket - subscription topic
+   */
+  initModalWsConnection = (websocket) => {
+    if (this.modalWsClient === null && websocket) {
+      this.modalWsClient = Stomp.Stomp.over(new SockJs(config.WS_URL));
```

Comment:
is it OK / optimal to open a new WS connection instead of using the current connection and just subscribing/unsubscribing from needed topics?

What's the technical reason behind this?

Reply:
Technical reason is not to mess with the logic from frontend/src/utils/websockets.js I really prefer to have a separate connection (that will be closed on modal closing) than to deal with the side effects that would be caused by having the the socket deactivated by other parts of the app and expect the data to come on a non existent conn. Hope it makes sense. I really want to have this separate.

Reply:
agreed with @petrican  to write here a link to the task where we discovered the buggy behaviour of WE/stomp library which constrains us to take this non-optimal approach.

Also, I understood if will be refactored later when we upgrade the WS/stomp libraries.

cc @siemiatj